ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -665,66 +680,41 @@ class JITModBuilder {
 
     // Store the active frames vector and its length in a separate struct to
     // save arguments.
-    AllocaInst *ActiveFramesStruct = storeVecWithLength(
-        FailBuilder, ActiveFrameVec,
-        ConstantInt::get(PointerSizedIntTy, ActiveFrames.size()));
+    AllocaInst *ActiveFramesStruct = createAndFillStruct(
+        FailBuilder, {ActiveFrameVec, ConstantInt::get(PointerSizedIntTy,
+                                                       ActiveFrames.size())});
+
+    // Make more space to store the locations of the corresponding live AOT
+    // values for this guard failure.
+    size_t Offset = LiveAOTSize;
+    LiveAOTSize += sizeof(AOTInfo) * LiveValues.size();
+    LiveAOTAddr = (AOTInfo *)realloc(LiveAOTAddr, LiveAOTSize);
```

Comment:
We should use `reallocarray(3)` here to avoid the possibility of a heap overflow.

Reply:
What's the advantage of this?

Reply:
It checks that `sizeof(AOTInfo) * LiveValues.size();` doesn't overflow, which can lead to memory errors.

Reply:
Fixed in [9f64524](https://github.com/ykjit/yk/pull/565/commits/9f64524a8c6ae37403c0a32a379e26fa9abd2270)